lib/internal/util/inspect.js

```diff
@@ -1542,6 +1542,33 @@ function reduceToSingleString(
   return `${braces[0]}${ln}${join(output, `,\n${indentation}  `)} ${braces[1]}`;
 }
 
+function hasBuiltInToString(value) {
+  // Count objects that have no `toString` function as built-in.
+  if (typeof value.toString !== 'function') {
+    return true;
+  }
+
+  let check = false;
+  let pointer = value;
+  // Find the object that has the `toString` property as own property in the
+  // prototype chain.
+  while (!hasOwnProperty(pointer, 'toString')) {
+    check = true;
+    pointer = Object.getPrototypeOf(pointer);
+  }
+
+  // Not a built-in.
+  if (!check) {
+    return false;
+  }
```

Comment:
Nit: what about unrolling it a little IIUC
```javascript
if (hasOwnProperty(value, 'toString'))
  return false;

let pointer = value;
do {
  pointer = Object.getPrototypeOf(pointer);
} while (!hasOwnProperty(pointer, 'toString'));
```

Reply:
I guess you mean because of the `check` variable`? This does not change the performance profile. Therefore I would keep it as is.